python/lsst/sconsUtils/state.py

```diff
@@ -131,6 +131,9 @@ def _initEnvironment():
       XPA_PORT
       CONDA_BUILD_SYSROOT
       SDKROOT
+      LD_PRELOAD
```

Comment:
This seems particularly dangerous, as it can cause unreproducible builds and tests if someone has this set to something unusual.  Is there any way to avoid this?

Reply:
LD_PRELOAD could only be passed if 
LD_PRELOAD=ldlogger.so and/or
CC_* variables are set

Reply:
👍 to the new code.